The Big Refactor

zysim opened the pull request:
I wanted to understand how `js/bingo.js` worked, so I decided to refactor it. And I got too carried away :^). First off, I'm not too sure if you'd like to consider accepting this PR? I'm more than happy to keep this on my own fork. That being said, to explain what I've done here; I've:

1. Split code up within `bingo.js` into smaller functions
2. Broke code for creating popouts out into its own file, `js/popout.js`
3. Converted both `bingo.js` and `tables/generic.js` into modules, and placed their imports in `index.html` to the bottom of `<body/>`
4. Written as much JSDoc as I possibly can :')
4. Changed `bingo-goals.txt` into `challenges.csv`
5. Changed the vocab around. E.g.:
    -  "goals" or "bingo" are now "challenges";
    - "synergy" is now "similarity"; etc.

I'm keeping this as a draft PR for now, as calculating the similarity of selected challenges is completely borked. The board still gets filled though, as this silently fails without breaking. Also, we still get empty cells here. Once I've fixed those, I'll turn this into a Big Boy PR.

Commit:
Fix Things

Commit:
More Refactoring

Commit:
More Refactoring

Commit:
About Done

Commit:
About There Now

Commit:
Add Todo to Readme

rekyuu commented:
Sounds like fun. The original code was based on the original bingo page by narcissa and I just filled in the blanks at the time. I tried opening the page before reading your comment so it looks like you're aware of the issues. Let me know when this is done and I can take a look at it. Let me know if I can help in any way.

zysim commented:
I'm pretty interested in the algo, actually. So I take it that Narcissa was the one who created it?

rekyuu commented:
As far as I know. I vaguely remember there being a generic board you could download and fill in the blanks but it's also entirely possible I just downloaded files from one of the bingo boards on this page: https://www.speedrunslive.com/tools

If there was a generic board, it's either been removed or I'm misremembering (probably the latter).

I have made my own bingo generator using Flask if you'd like to look. Difference is the values are completely random instead of rated on category or difficulty. The values are just stored in an array in the controller, chosen at random, then dumped into the page template. 

The only downside to this is you'd have to host it yourself somewhere instead of hosting it on GitHub pages, but you get better flexibility with your code stack. I'll probably dockerize it at some point.

https://github.com/rekyuu/horror_bingo

Personally I think it would be more interesting (and fun) to make a new bingo framework but we already have stuff like BingoSync so I'm not sure how necessary or feasible a project would be.

zysim commented:
Ah, Flask. I'll check your project out :). Yeah I've got no prob with having to set a server up and stuff; the plan (for me) now's to migrate the bingo board to _probably_ Auride's GH pages, if not to my own domain + server setup.

And yeah tbh I feel like there's no point recreating the board from scratch here haha. I've got no energy to wanna do so anyway :P

Your bingo list is _good_

Commit:
Start From Scratch

Commit:
Restart

Commit:
Think It's Good

Commit:
Made Card Buttons Work Again

Commit:
Rename 'types' to 'cats'

Commit:
Docs and Reorders

Commit:
Extract Bingo Footer Code

zysim commented:
I restarted from scratch. It should work the same way now still.

Commit:
Styles

Commit:
Get Rid of Classes for Header and Footer

Commit:
Slim Down Algo
Instead of groups of challenges with cats, the data source is simply an
array of objects with "name" as its sole key. This also means the algo
will not have difficulty and similarity calculations. We simply choose a
random set of 25 challenges.

Commit:
Remove 'mode' Setting

rekyuu commented:
Hey, I kind of forgot about this. Is it ready for review?

zysim commented:
This is a very good question. I need to refresh my memory on this PR tbh; I think there actually wasn't a point in this PR in the end lmao

zysim commented:
Well, so it's _ready_, but I realise I've changed quite a lot, so I'm not sure if it's worth your time to go through it? If you'd like to still, feel free to, of course :)

dshepsis committed:
Source objectives from Google Sheets (#1)

* Removing some unecessary things and adding/editing/removing objectives

* Using google sheets api

I changed the code so it loads objectives from this Google Sheets
spreadsheet:
https://docs.google.com/spreadsheets/d/1eTmfECVVlp1nswOpbsviUjMK-cUwHziLK5nLn4rbuH4/edit?usp=sharing

* Removing some unecessary things, changing getChallenges algo
